lib/grinder_sdk.dart

```diff
@@ -339,6 +352,41 @@ class Analyzer {
     run_lib.run(_sdkBin('dartanalyzer'), arguments: args);
   }
 
+  static List<String> _expandDirectoriesToDartFiles(List<String> paths) {
+    List<String> files = [];
+
+    paths.forEach((p) {
+      if (FileSystemEntity.typeSync(p) == FileSystemEntityType.DIRECTORY) {
+        files.addAll(_collectDartFiles(new Directory(p)));
+      } else {
+        files.add(p);
+      }
+    });
+    return files;
+  }
+
+  static List<String> _collectDartFiles(Directory directory) {
+    var files = <String>[];
+    if (directory.existsSync()) {
+      for (var entry
+          in directory.listSync(recursive: true, followLinks: false)) {
+        //var relative = path.relative(entry.path, from: directory.path);
+
+        if (_isDartFileName(entry.path) && !_isInHiddenDir(directory.path)) {
+          files.add(entry.path);
+        }
+      }
+    }
+    return files;
+  }
+
+  /// Returns [:true:] if this [fileName] is a Dart file.
```

Comment:
[:true:]

is the legacy format, and is not standard markdown.  I prefer backticks:

    `true`